src/cassiopeia/utils.py

```
@@ -180,3 +181,54 @@ def _combine_edge_data(parent_edge: dict[str, Any], child_edge: dict[str, Any])
         child_length = child_edge.get("length", 0)
         new_edge["length"] = parent_length + child_length
     return new_edge
+
+
+def _get_character_matrix(tree, characters_key: str = "characters") -> np.ndarray:
```

Should this function raise a KeyError if characters_key not present?

---

Does this automatically raise a KeyError if the characters_key isn't there?